### ISSUE TITLE
direnv export json - not producing any output

### ISSUE THREAD

**Issue by fiksn** [direnv/direnv#809]
**Describe the bug**

I've stumbled on this when investigating https://github.com/fehnomenal/intellij-direnv/issues/11

Might just be me not understanding how that is supposed to work. (Apologies if that is the case.)

**To Reproduce**
`direnv export json` in workdir containing .envrc with
```
export A=blah
```
should produce some output I believe? It does not matter whether I do `direnv allow .` before or not.
I've tried to set `export DIRENV_DEBUG=1` and then I can see
```
direnv: export:12:03:41 cmd_export.go:22: start
direnv: export:12:03:41 cmd_export.go:35: loading RCs
direnv: export:12:03:41 cmd_export.go:43: updating RC
direnv: export:update:12:03:41 cmd_export.go:46: Determining action:
...
direnv: export:update:12:03:41 cmd_export.go:60: no update needed
```

**Environment**
- macOS Catalina 10.15.7
- zsh
- direnv 2.22.0

**Comment by fiksn** [comment]
Note: tried on some old machine. Same issue is happening also with:

macOS BigSur 11.4
zsh 5.8 (x86_64-apple-darwin20.0)
direnv 2.28.0

**Comment by zimbatm** [comment]
`direnv export` emits the environment diff that should be applied to the parent process. So if the environment was already loaded, the export will be empty.

One trick is to unload the environment with `direnv exec / <command>` and then chain it: `direnv exec / direnv export json`

**Comment by fiksn** [comment]
Thanks for the explanation, learned something new today. Apparently `direnv export` is internally used as well, so it makes sense that it behaves like this. I've noticed that on Linux `direnv exec /` actually worked while on mac I had to point it to some dir. containing a dummy `.envrc` file in order to get the "chain" working. 

I have to admit my mistake - this is not a `direnv` bug. Would have been too easy if something as simple as this was causing my IDE to not work with `.envrc`.
